app.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Range) || is_a?(Hash)
+    while i < new_array.length
+      yield new_array[i]
+      i += 1
+    end
+    new_array
+  end
```

Comment:
# my_each
1. [ ] When a block is given when self is a range returns the range itself after calling the given block once for each element in self
### Test 👉 `range.my_each(&block)`
|Expected|Got|
|:---:|:---:|
|`5..50`|` [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]`|